enterprise/coderd/templates.go

```diff
@@ -163,14 +163,18 @@ func (api *API) patchTemplateACL(rw http.ResponseWriter, r *http.Request) {
 			}
 		}
 
-		template, err = tx.UpdateTemplateACLByID(ctx, database.UpdateTemplateACLByIDParams{
+		err = tx.UpdateTemplateACLByID(ctx, database.UpdateTemplateACLByIDParams{
 			ID:       template.ID,
 			UserACL:  template.UserACL,
 			GroupACL: template.GroupACL,
 		})
 		if err != nil {
 			return xerrors.Errorf("update template ACL by ID: %w", err)
 		}
+		template, err = tx.GetTemplateByID(ctx, template.ID)
+		if err != nil {
+			return xerrors.Errorf("get updated template by ID: %w", err)
+		}
```

Comment:
Yea, it is unfortunate. I could have made the audit log table take `TemplateTable`, but then we would have 2 types floating around.

We do have 1 less read in the fetch case now as we do not need to fetch the user, so it's a trade for 1 call on the read, to 1 call on the update.

If we are willing to add postgres `RULE`s we can actually "insert into a view". But that would complicate the view.

Reply:
> it's a trade for 1 call on the read, to 1 call on the update.
I think this is a fair trade.

Reply:
> It would be nice if we could move this into a trigger on the audit_logs table.

Unfortunately we support multiple paths for audit logs to be exported, so it can't just be as simple as a trigger :(